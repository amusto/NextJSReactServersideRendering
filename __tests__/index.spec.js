import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

describe('With Enzyme', () => {
  it('App shows "Submit a Message!"', () => {
    const app = shallow(<App />)

    expect(app.find('h2').text()).toEqual('Submit a Message!')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Submit a Message!"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})