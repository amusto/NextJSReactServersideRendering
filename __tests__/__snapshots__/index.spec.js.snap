// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div
    className="jsx-3094272964"
  >
    <a
      className="jsx-3094272964"
      href="/"
      onClick={[Function]}
    >
      Add a Guest
    </a>
    <a
      className="jsx-3094272964"
      href="/guestbook"
      onClick={[Function]}
    >
      Guestbook
    </a>
  </div>
  <div
    className="jsx-1758071747"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <form
      className="jsx-1758071747"
      onSubmit={[Function]}
    >
      <label
        className="jsx-1758071747"
        htmlFor="fullName"
      >
        Full Name
      </label>
      <input
        className="jsx-1758071747"
        name="fullname"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your full name"
        type="text"
        value=""
      />
      <label
        className="jsx-1758071747"
        htmlFor="fullname"
      >
        Message
      </label>
      <input
        className="jsx-1758071747"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a message"
        type="text"
        value=""
      />
      <input
        className="jsx-1758071747"
        disabled={false}
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;

exports[`With Snapshot Testing App shows "Submit a Message!" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div
    className="jsx-3094272964"
  >
    <a
      className="jsx-3094272964"
      href="/"
      onClick={[Function]}
    >
      Add a Guest
    </a>
    <a
      className="jsx-3094272964"
      href="/guestbook"
      onClick={[Function]}
    >
      Guestbook
    </a>
  </div>
  <h2
    className="jsx-83119193"
  >
    Submit a Message!
  </h2>
  <div
    className="jsx-1758071747"
    style={
      Object {
        "margin": "10px",
      }
    }
  >
    <form
      className="jsx-1758071747"
      onSubmit={[Function]}
    >
      <label
        className="jsx-1758071747"
        htmlFor="fullName"
      >
        Full Name
      </label>
      <input
        className="jsx-1758071747"
        name="fullname"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter your full name"
        type="text"
        value=""
      />
      <label
        className="jsx-1758071747"
        htmlFor="fullname"
      >
        Message
      </label>
      <input
        className="jsx-1758071747"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a message"
        type="text"
        value=""
      />
      <input
        className="jsx-1758071747"
        disabled={false}
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</div>
`;
